packages/app-builder-lib/src/macPackager.ts

```diff
@@ -192,8 +181,10 @@ export default class MacPackager extends PlatformPackager<MacConfiguration> {
       }
     }
 
-    if (nonMasPromise != null) {
-      await nonMasPromise
+    if ((!hasMas || targets.length > 1) && !prepackaged) {
+      const appPath = path.join(this.computeAppOutDir(outDir, arch), `${this.appInfo.productFilename}.app`);
+      await this.doPack(outDir, path.dirname(appPath), this.platform.nodeName as ElectronPlatformName, arch, this.platformSpecificBuildOptions, targets)
+      .then(() => this.packageInDistributableFormat(appPath, arch, targets, taskManager))
```

Comment:
This logic is no different than before, right? It's simply been moved.

Creating `nonMasPromise` initially does not actually execute it until we run `await nonMasPromise` here. So moving that logic from line 164 to here (line 195) by simply not preallocating the promise does nothing different in logic flow.

Not sure how this fixes the issue you're describing.

Reply:
Try replacing this logic with this 👇 . I think this replicates previous logic. From what I can tell, the `.then(() => this.packIndDistributableFormat(...))` is the issue?
```
    if (!hasMas || targets.length > 1) {
      const appPath = prepackaged == null ? path.join(this.computeAppOutDir(outDir, arch), `${this.appInfo.productFilename}.app`) : prepackaged
      if (prepackaged == null) {
        await this.doPack(outDir, path.dirname(appPath), this.platform.nodeName as ElectronPlatformName, arch, this.platformSpecificBuildOptions, targets)
      }
      this.packageInDistributableFormat(appPath, arch, targets, taskManager)
    }
```